pkg/str/str.go

```diff
@@ -0,0 +1,26 @@
+// Package str contains string utilities.
+package str
+
+// In returns true if string v is one of s strings.
+func In(v string, s ...string) bool {
+	for _, x := range s {
+		if x == v {
+			return true
+		}
+	}
+	return false
+}
+
+// Uniq returns a new slice containing only unique strings.
+// The order of the return value is undefined.
+func Uniq(strs ...string) []string {
+	m := make(map[string]struct{}, len(strs))
+	for _, s := range strs {
+		m[s] = struct{}{}
+	}
+	out := make([]string, 0, len(m))
```

Comment:
If I understand correctly, the first two allocate memory for a string slice with zero elements. What does the third parameter do here?

Reply:
The third arg makes sure the string slice is backed by an underlying array with a capacity of `len(m)`. Since I know what the final size of the slice will be at the end, this prevents any reallocation/copying when I do `out = append(out, s)` later. It's a minor performance improvement.

https://blog.golang.org/go-slices-usage-and-internals

Reply:
Makes sense. Thank you!